qiskit/providers/aer/extensions/snapshot.py

```diff
@@ -0,0 +1,147 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2018, 2019.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""
+Simulator command to snapshot internal simulator representation.
+"""
+
+import warnings
+
+from qiskit import QuantumCircuit
+from qiskit.circuit.quantumregister import QuantumRegister
+from qiskit.circuit import Instruction
+from qiskit.extensions.exceptions import QiskitError, ExtensionError
+
+class Snapshot(Instruction):
+    """Simulator snapshot instruction."""
+
+    def __init__(self,
+                 label,
+                 snapshot_type='statevector',
+                 num_qubits=0,
+                 num_clbits=0,
+                 params=None):
+        """Create new snapshot instruction.
+        Args:
+            label (str): the snapshot label for result data.
+            snapshot_type (str): the type of the snapshot.
+            num_qubits (int): the number of qubits for the snapshot type [Default: 0].
+            num_clbits (int): the number of classical bits for the snapshot type [Default: 0].
+            params (list or None): the parameters for snapshot_type [Default: None].
+        Raises:
+            ExtensionError: if snapshot label is invalid.
+        """
+        if not isinstance(label, str):
+            raise ExtensionError('Snapshot label must be a string.')
+        self._label = label
+        self._snapshot_type = snapshot_type
+        if params is None:
+            params = []
+        super().__init__('snapshot', num_qubits, num_clbits, params)
+
+    def assemble(self):
+        """Assemble a QasmQobjInstruction"""
+        instruction = super().assemble()
+        instruction.label = self._label
+        instruction.snapshot_type = self._snapshot_type
+        return instruction
+
+    def inverse(self):
+        """Special case. Return self."""
+        return Snapshot(self.num_qubits, self.num_clbits, self.params[0],
+                        self.params[1])
+
+    @staticmethod
+    def define_snapshot_register(self,
```

Comment:
Can you rename `self` to `circuit` or similar? As this is a static method it doesn't take an instance of the class it belongs to, which is what `self` usually means.